scalastyle-config.xml

```diff
@@ -155,6 +155,14 @@ This file is divided into 3 sections:
     <customMessage><![CDATA[MDC should be used in string interpolation log"..." instead of s"..."]]></customMessage>
   </check>
 
+  <check customId="logInlineVariable" level="error" class="org.scalastyle.file.RegexChecker" enabled="true">
+    <parameters><parameter name="regex">log(?:Info|Warning|Error)\(s".*(\$|\+\s*[^\s"]).*"\)</parameter></parameters>
```

Comment:
we need to detect the ".format(..)` as well.
Also, need to exclude the testing code.

Reply:
Similar rules also need to be added to `dev/checkstyle.xml` to check Java code

Reply:
> Also, need to exclude the testing code.

This might be difficult, as for scalastyle-maven-plugin, there is currently no mechanism similar to Java checkstyle's SuppressionFilter that can ignore certain rules in specific files or code blocks. The rule ignoring of scalastyle is global and cannot be targeted at specific files or code blocks. After defining the check rule, we may only be able to skip the relevant checks in a way similar to the following:

```
// scalastyle:off logInlineVariable
logInfo(s"....")
// scalastyle:on logInlineVariable
```